aider/models/model.py

```diff
@@ -0,0 +1,52 @@
+saved_openai = None
+
+
+class Model:
+    name = None
+    edit_format = None
+    max_context_tokens = 0
+    tokenizer = None
+
+    always_available = False
+    use_repo_map = False
+    send_undo_reply = False
+
+    prompt_price = None
+    completion_price = None
+
+    def __init__(self, name, openai=None):
+        global saved_openai
+        if (openai and "openrouter.ai" in openai.api_base):
+            saved_openai = openai
```

Comment:
Can you help me understand the role of `saved_openai` ?

Reply:
I don't love the approach but the problem I was trying to solve was how Model calls which didn't have openai passed in (i.e. not called on main.py:482) would be able to check for openrouter.ai in the api_base - in particular how can the model calls for chat summaries or git commits remember to use the openrouter url instead of the default one.